### PR TITLE
[FLINK-27290][docs-zh]Translate the "Data Type" section of "Data Types" in to Chinese.

### PR DESCRIPTION
**Pull request by ChengkaiYang2022**
## What is the purpose of the change

Translate the "Data Type" section of "Data Types" in to Chinese.

## Verifying this change

Please make sure both new and modified tests in this PR follows the conventions defined in our code quality guide: https://flink.apache.org/contributing/code-style-and-quality-common.html#testing

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (100MB)*
  - *Extended integration test for recovery after master (JobManager) failure*
  - *Added test that validates that TaskInfo is transferred only once across recoveries*
  - *Manually verified the change by running a 4 node cluster with 2 JobManagers and 4 TaskManagers, a stateful streaming program, and killing one JobManager and two TaskManagers during the execution, verifying that recovery happens correctly.*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / no) no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / no) no
  - The serializers: (yes / no / don't know) no
  - The runtime per-record code paths (performance sensitive): (yes / no / don't know) no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / no / don't know) no 
  - The S3 file system connector: (yes / no / don't know) no
 
## Documentation

  - Does this pull request introduce a new feature? (yes / no) no
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by ChengkaiYang2022** [comment]
datatype

**Comment by ChengkaiYang2022** [comment]
@fsk119 Hi, would you mind to review this PR?

File: docs/content.zh/docs/dev/table/types.md [hunk 1]
```diff
@@ -24,65 +24,65 @@ specific language governing permissions and limitations
...
+可能会使用到 Python API 中基于 `pyflink.table.types.DataType` 的一些实例。
 
-A `DataType` instance has such a responsibility:
-- **Declaration of a logical type** which does not imply a concrete physical representation for transmission
-or storage but defines the boundaries between Python languages and the table ecosystem.
+`数据类型` 实例有如下职责：
+- **作为逻辑类型的表现形式**，定义 JVM 类语言或 Python 语言与 Table 生态系统的边界，而不是以具体的物理表现形式存在于数据的传输过程或存储中。
 
-For Python language, those types are available in `pyflink.table.types.DataTypes`.
+对于 Python 语言，这些数据类型可以在 `pyflink.table.types.DataTypes` 下找到。
```
Comment (deadwind4) [inline docs/content.zh/docs/dev/table/types.md:71]:
```suggestion
对于 Python 语言，这些类型可以在 `pyflink.table.types.DataTypes` 下找到。
```

File: docs/content.zh/docs/dev/table/types.md [hunk 2]
```diff
@@ -100,47 +100,38 @@ t = DataTypes.INTERVAL(DataTypes.DAY(), DataTypes.SECOND(3))
...
-expects.
+在Table 生态系统中，当需要将 SQL 中的数据类型对应到实际编程语言中的数据类型时，就需要有物理提示。物理提示明确了对应过程中应该使用哪种数据格式。
 
-For example, a data source could express that it produces values for logical `TIMESTAMP`s using a `java.sql.Timestamp` class
-instead of using `java.time.LocalDateTime` which would be the default. With this information, the runtime is able to convert
-the produced class into its internal data format. In return, a data sink can declare the data format it consumes from the runtime.
+比如，在 source 端产生数据时，可以规定：`TIMESTAMP` 的逻辑类型，在底层要使用 `java.sql.Timestamp` 这个类表示，而不是使用默认的 `java.time.LocalDateTime` 类。有了物理提示，可以帮助 Flink 运行时根据提供的类将数据转换为其内部数据格式。同样在 sink 端，定义好数据格式，以便能从 Flink 运行时获取、转换数据。
 
-Here are some examples of how to declare a bridging conversion class:
+下面的例子展示了如何使用 Bridging 的方式指定要转换的类：
```
Comment (deadwind4) [inline docs/content.zh/docs/dev/table/types.md:111]:
```suggestion
下面的例子展示了如何声明一个桥接转换类：
```

File: docs/content.zh/docs/dev/table/types.md [hunk 3]
```diff
@@ -100,47 +100,38 @@ t = DataTypes.INTERVAL(DataTypes.DAY(), DataTypes.SECOND(3))
...
+// 告诉 Flink 运行时使用基本的 int 数组来处理数据，而不是用包装类 Integer 数组
 val t: DataType = DataTypes.ARRAY(DataTypes.INT().notNull()).bridgedTo(classOf[Array[Int]])
 ```
 {{< /tab >}}
 {{< /tabs >}}
 
-<span class="label label-danger">Attention</span> Please note that physical hints are usually only required if the
-API is extended. Users of predefined sources/sinks/functions do not need to define such hints. Hints within
-a table program (e.g. `field.cast(TIMESTAMP(3).bridgedTo(Timestamp.class))`) are ignored.
+<span class="label label-danger">注意</span> 请记住，只有在需要扩展 API 时才需要使用到物理提示。使用预定义的 source、sink 以及 Flink 函数时，不需要用到物理提示。在使用 Table API 编写程序时，Flink 会忽略物理提示（例如 `field.cast(TIMESTAMP(3).bridgedTo(Timestamp.class))`）。
```
Comment (deadwind4) [inline docs/content.zh/docs/dev/table/types.md:134]:
```suggestion
<span class="label label-danger">注意</span> 请记住，只有在扩展 API 时才需要使用到物理提示。使用预定义的 source、sink 以及 Flink 函数时，不需要用到物理提示。在使用 Table API 编写程序时，Flink 会忽略物理提示（例如 `field.cast(TIMESTAMP(3).bridgedTo(Timestamp.class))`）。
```

**Comment by ChengkaiYang2022** [comment]
Anyway, I will correct the git message by adding [docs-zh] when I rebase.

**Review by MartijnVisser — APPROVED** [review]
@ChengkaiYang2022  @deadwind4 Thank you both. LGTM

**Pull request merged**